crates/holochain_conductor_api/src/config/conductor/dpki_config.rs

```diff
@@ -77,11 +79,17 @@ impl DpkiConfig {
 
 impl Default for DpkiConfig {
     fn default() -> Self {
-        Self {
-            dna_path: None,
-            network_seed: DPKI_NETWORK_SEED_TESTING.to_string(),
-            allow_throwaway_random_dpki_agent_key: false,
-            no_dpki: false,
+        cfg_if! {
+            if #[cfg(feature = "dpki")] {
+                Self {
+                    dna_path: None,
+                    network_seed: DPKI_NETWORK_SEED_TESTING.to_string(),
+                    allow_throwaway_random_dpki_agent_key: false,
+                    no_dpki: false,
+                }
+            } else {
+                Self::disabled()
```

Comment:
We should really also be testing that this behaves as expected. Something like, a test which explicitly configures DPKI on without the feature and then asserts that it is disabled.

Should we not also be printing a warning somewhere if that happens? So the user knows to turn the feature on if they want DPKI?

This complicated by testing with the feature turned on. We should really be running tests against different combinations of features. Most projects do. That's not a problem for this PR but if we can add a test which is behind a `cfg(not(feature = "dpki"))` then I'm happy to open a follow-up ticket to look at our CI

Reply:
What happens now if the user has the feature disabled and explicitly configures DPKI on? That should be rejected with a warning outside of tests too right?